bin/proving-service/src/commands/worker.rs

```diff
@@ -6,6 +6,39 @@ use tracing::{info, instrument};
 
 use crate::{api::RpcListener, generated::api_server::ApiServer, utils::MIDEN_PROVING_SERVICE};
 
+/// Defines the possible types of provers that a worker can be.
+#[derive(Debug, Parser, Clone, Default)]
```

Comment:
We could derive `Copy` since this is just two (or later three?) bools.

Nice solution integrating it directly into clap btw :rocket:.